### PR TITLE
Reduce stores coupling to FileSystemAbstraction

### PR DESCRIPTION
**Pull request by chrisvest**
This moves us closer to supporting block device storage.

### CODE REVIEW

**Comment by chrisvest** [comment]
@tinwelint Can you take a look at the file locking changes?

**Comment by digitalstain** [comment]
@chrisvest Needs rebase and perhaps some move verbosity of the commit messages. I'll label as not ready.

File: community/io/src/test/java/org/neo4j/io/pagecache/impl/SingleFilePageSwapperTest.java [hunk 1]
```diff
@@ -20,24 +20,44 @@
...
 import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
 import static org.neo4j.test.ByteArrayMatcher.byteArray;
 
 public class SingleFilePageSwapperTest extends PageSwapperTest
 {
+    @Rule
+    public final TargetDirectory.TestDirectory testDir = TargetDirectory.testDirForTest( getClass() );
+
```
Comment (chrisvest) [inline community/io/src/test/java/org/neo4j/io/pagecache/impl/SingleFilePageSwapperTest.java]: No, PageSwapperTest already has a `testDir` from a `TestDirectory` rule.

**Comment by chrisvest** [comment]
@digitalstain Done.

**Comment by digitalstain** [comment]
This PR depends on this: https://github.com/neo4j/neo4j/pull/5024 to properly initialize the fields of the NeoStore file during verification.

Also, it changes the order in which the index names are returned from GDS.index().nodeIndexNames() which makes StoreMigratorFrom19IT break in verifyDatabaseContents() when it verifies the legacy indexes - the names are correct but in the wrong order. It does not look like to be a bug, so simply amending that test should do it.

**Pull request merged**